### PR TITLE
EDL fix for invalid IPs

### PR DESCRIPTION
**Pull request by PA-Rweins**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/44526)

## Description
- Before collapsing to either CIDR or an IP range each IP address or CIDR is validated.
- Invalid IPs/CIRDs will be shown as part of the list but will not be collapsed.

## Screenshots
Screen Shot 2021-12-16 at 17 43 56

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 1]
```diff
@@ -216,14 +217,51 @@ def ips_to_ranges(ips: Iterable, collapse_ips: str):
...
+    if collapse_ips == COLLAPSE_TO_RANGES:
+        ips_range_groups = IPSet(valid_ips).iter_ipranges()
+        collapsed_list = ip_groups_to_ranges(ips_range_groups)
     else:
-        cidrs = IPSet(ips).iter_cidrs()
-        return ip_groups_to_cidrs(cidrs)
+        cidrs = IPSet(valid_ips).iter_cidrs()
+        collapsed_list = ip_groups_to_cidrs(cidrs)
+
+    collapsed_list.update(invalid_ips)
```
Comment (Bargenish) [inline Packs/EDL/Integrations/EDL/EDL.py:236]: Why do we still want to add the invalid ips?

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -216,14 +217,51 @@ def ips_to_ranges(ips: Iterable, collapse_ips: str):
...
+    return collapsed_list
+
+
+def is_valid_ip(ip: str) -> bool:
```
Comment (Bargenish) [inline Packs/EDL/Integrations/EDL/EDL.py:240]: let's check the structure and only have maximum, 2 checks per IP.
not sure how scalable it is.
@DeanArbel wdyt?

**Review by Bargenish — CHANGES_REQUESTED** [review]

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 1 above)
Comment (DeanArbel) [reply Packs/EDL/Integrations/EDL/EDL.py:236]: It's a valid usecase. Could be a valid indicator that was created with a wrong type, so it should be in the EDL, but not calculated under collapsible ips.

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 2 above)
Comment (DeanArbel) [reply Packs/EDL/Integrations/EDL/EDL.py:240]: I agree.
I think it's better to drop the `generic` implementation of `ip_groups_to_ranges` and `ips_to_ranges` by passing an extra argument - the type of expected `IP` / `CIDR` - that way the code doesn't need to be made of 4 `try/except`.

Comment (PA-Rweins) [reply Packs/EDL/Integrations/EDL/EDL.py:240]: Fixed

**Review by DeanArbel — APPROVED** [review]
Good job

**Review by Bargenish — APPROVED** [review]

**Pull request merged**